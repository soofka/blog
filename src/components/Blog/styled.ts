import styledComponents from 'styled-components';

export const StyledBlog = styledComponents.div`
  width: 100%;
  height: 100%;
  font-size: ${props => props.theme.baseFontSize};
  line-height: ${props => props.theme.baseLineHeight};
`;

export const StyledBlogContainer = styledComponents.div`
  position: relative;
  width: 100%;
  max-width: ${props => props.theme.siteMaxWidth};
  height: 100%;
  margin: 0 auto;
`;
