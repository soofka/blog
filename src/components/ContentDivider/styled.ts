import styledComponents from 'styled-components';

export const StyledContentDivider = styledComponents.hr`
  border-color: ${props => props.theme.lighterBlue} !important;
`;

export default StyledContentDivider;
