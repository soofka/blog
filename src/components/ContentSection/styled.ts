import styledComponents from 'styled-components';

export const StyledContentSection = styledComponents.div`
  float: left;
  margin: 0 0 0 ${props => props.theme.infoSectionWidth};
  padding: ${props => props.theme.paddingM};
  width: 100%;
  height: auto;
  min-height: 100%;
  box-sizing: border-box;
  color: ${props => props.theme.lighterBlack};
  background-color: ${props => props.theme.white};
  box-shadow: ${props => `0 0 ${props.theme.baseDistanceValue}px 0 ${props.theme.white}`};

  a {
    color: ${props => props.theme.lighterBlack};

    &:hover,
    &:active {
      color: ${props => props.theme.black};
    }
  }

@media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
  margin: 0;
  padding: ${props => props.theme.paddingS};
  width: 100%;
}`;

export default StyledContentSection;
