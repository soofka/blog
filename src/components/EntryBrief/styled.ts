import styledComponents from 'styled-components';

export const StyledEntryBrief = styledComponents.div`
  float: left;
  width: 100%;
  height: ${props => 24 * props.theme.baseDistance}px;
  margin: ${props => props.theme.marginS} 0 ${props => props.theme.marginM} 0;
  padding: 0;
  border: 2px solid ${props => props.theme.white};
  outline: 1px solid ${props => props.theme.lighterBlack};
  background-repeat: no-repeat;
  background-position: center center;
  background-attachment: scroll;
  background-size: cover;
`;

export default StyledEntryBrief;
