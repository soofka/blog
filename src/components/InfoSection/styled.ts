import styledComponents from 'styled-components';

export const StyledInfoSection = styledComponents.div`
  position: static;
  display: inline-block;
  width: 100%;
  height: ${props => props.theme.infoSectionHeightValue}px;
  color: ${props => props.theme.darkerWhite};
  background-color: ${props => props.theme.blue};
  box-shadow: ${props => `0 0 ${props.theme.baseDistanceValue}px 0 ${props.theme.darkerBlue}`};

  a {
    color: ${props => props.theme.darkerWhite};
    text-decoration: none;

    &:hover,
    &:active {
      color: ${props => props.theme.white};
    }
  }
`;

// export const StyledInfoSection = styledComponents.div`
//   position: absolute;
//   top: 0;
//   left: -${props => props.theme.infoSectionWidthValue}px;
//   box-sizing: border-box;
//   width: ${props => props.theme.infoSectionWidthValue}px;
//   height: auto;
//   min-height: 100%;
//   padding: ${props => props.theme.paddingS};
//   color: ${props => props.theme.darkerWhite};
//   background-color: ${props => props.theme.blue};
//   box-shadow: ${props => `0 0 ${props.theme.baseDistanceValue}px 0 ${props.theme.darkerBlue}`};
//
//   a {
//     color: ${props => props.theme.darkerWhite};
//     text-decoration: none;
//
//     &:hover,
//     &:active {
//       color: ${props => props.theme.white};
//     }
//   }
//
//   @media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
//     position: static;
//     display: inline-block;
//     width: 100%;
//     height: ${props => props.theme.infoSectionMobileHeightValue}px;
//     min-height: 0;
// }
// `;

export default StyledInfoSection;

