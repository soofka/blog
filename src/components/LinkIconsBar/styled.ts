import styledComponents from 'styled-components';

export const StyledLinkIconsBar = styledComponents.ul`
  margin: ${props => props.theme.marginM};
  padding: 0;
  text-align: center;

  li {
    display: inline-block;
    margin: ${props => `${props.theme.marginM} ${props.theme.marginXS}`};
    padding: 0;
    font-size: ${props => props.theme.fontSizeXXL};
    line-height: ${props => props.theme.fontSizeXXL};

    svg {
      width: 40px;
      height: 40px;
      opacity: 0.7;

      &:hover,
      &:active {
        opacity: 1;
      }
    }
  }

  @media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
    float: right;
    margin: ${props => props.theme.marginS} 0;

    li {
      margin: 0 ${props => props.theme.marginXS};
      padding: 0;
      font-size: ${props => props.theme.fontSizeL};
      line-height: ${props => props.theme.fontSizeL};
    }
  }
`;

export default StyledLinkIconsBar;
