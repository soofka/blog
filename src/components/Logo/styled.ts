import styledComponents from 'styled-components';

export const StyledLogo = styledComponents.img`
  display: block;
  margin: ${props => props.theme.marginL} auto;
  padding: 0;
  width: ${props => 28 * props.theme.baseDistance}px;
  height: ${props => 28 * props.theme.baseDistance}px;
  border-radius: ${props => 4 * props.theme.baseDistance}px;

  @media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
    float: left;
    margin: 0;
    padding: 0;
    width: ${props => 8 * props.theme.baseDistance}px;
    height: ${props => 8 * props.theme.baseDistance}px;
    border-radius: ${props => props.theme.baseDistance}px;
  }
`;

export default StyledLogo;
