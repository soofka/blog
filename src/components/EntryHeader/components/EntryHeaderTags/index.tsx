import * as React from 'react';
import { Link } from 'react-router-dom';

import RoutingProvider from 'common/RoutingProvider';

import { StyledEntryHeaderTags } from './styled';

interface EntryHeaderTagsPropsInterface {
  tags: string[];
}

export const EntryHeaderTags = (props: EntryHeaderTagsPropsInterface): JSX.Element => {
  const { tags } = props;

  return (
    <StyledEntryHeaderTags>
      {tags.map((tag: string, index: number) => {
        return (
          <Link
            key={index}
            to={`${RoutingProvider.getEntriesByTagNiceUrlBasePath()}/${RoutingProvider.parseTextToNiceUrl(tag)}`}
          >
            {tag}
          </Link>
        );
      })}
    </StyledEntryHeaderTags>
  );
};

export default EntryHeaderTags;
