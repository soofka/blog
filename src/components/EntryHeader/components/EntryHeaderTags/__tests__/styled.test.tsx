import * as React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import theme from 'common/theme';
import { StyledEntryHeaderTags } from 'components/EntryHeader/components/EntryHeaderTags/styled';

describe('<StyledEntryHeaderTags/>', () => {

  it('is styled correctly', () => {
    expect(shallow(<StyledEntryHeaderTags theme={theme} />)).toMatchSnapshot();
  });

});
