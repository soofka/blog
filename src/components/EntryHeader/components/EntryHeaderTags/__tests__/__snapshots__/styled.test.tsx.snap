// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledEntryHeaderTags/> is styled correctly 1`] = `
.c0 {
  font-size: 0.7em;
  line-height: 1.75em;
  margin: 0 0 16px 0;
  padding: 0;
}

.c0 a {
  display: inline-block;
  margin: 2px;
  padding: 4px 8px;
  background-color: rgb(222,222,222);
  color: rgb(0,1,49);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:hover {
  background-color: rgb(198,198,198);
  color: rgb(0,0,24);
}

<p
  className="c0"
/>
`;
